src/components/VerticalBarChart.tsx

```diff
@@ -130,7 +132,77 @@ export default function VerticalBarChart({
       .attr("x", (d) => x(d.year) ?? 0)
       .attr("y", (d) => y(d.value))
       .attr("height", (d) => y(0) - y(d.value))
-      .attr("width", x.bandwidth());
+      .attr("width", x.bandwidth())
+      .on("mouseover", onMouseOver)
+      .on("mouseout", onMouseOut);
+
+    // Update tooltips
+    const ttwidth = 110;
+
+    select(tooltips.current)
+      .selectAll<SVGPathElement, unknown>("path")
+      .data(d3data)
+      .join("path")
+      .attr("display", "none")
+      .attr("fill", "white")
+      .attr("stroke", "black")
+      .attr("stroke-width", 1)
+      .attr("stroke-linejoin", "round")
+      .attr(
+        "transform",
+        (d) =>
+          "translate(" +
+          (x(d.year) + x.bandwidth() / 2) +
+          " " +
+          y(d.value) +
+          ")",
+      )
+      .attr(
+        "d",
+        "M0,0 l 5,-5 h " +
+          (ttwidth / 2 - 5) +
+          " v -20 h -" +
+          ttwidth +
+          " v 20 h " +
+          (ttwidth / 2 - 5) +
+          " Z",
+      );
+
+    select(tooltips.current)
+      .selectAll<SVGTextElement, string>("text")
+      .data(d3data)
+      .join("text")
+      .attr("display", "none")
+      .attr("x", (d) => x(d.year) + x.bandwidth() / 2)
+      .attr("y", (d) => y(d.value) - 10)
+      .attr("text-anchor", "middle")
+      .attr("font-size", "12px")
+      .text((d) => d.value + " " + d.unit);
+
+    function setTooltipDisplay(year: string | null) {
+      select(tooltips.current)
+        .selectAll<SVGTextElement, string>("text")
+        .join()
+        .attr("display", (d) =>
+          year !== null && d.year === year ? "display" : "none",
+        );
+
+      select(tooltips.current)
+        .selectAll<SVGPathElement, unknown>("path")
+        .join()
+        .attr("display", (d) =>
+          year !== null && d.year === year ? "display" : "none",
+        );
+    }
+
+    function onMouseOver(event) {
```

Comment:
@cjyetman I think if you add a type to this `event` (Not sure offhand what the right type is though), that should satisfy eslint that it does in fact have a `.currentTarget` member.

Reply:
there's also a way to weakly type, I believe, where you just specify that event has the .currentTarget (and you don't care about anything else)

Reply:
sounds good, but I don't know how to do either of those options 🤦🏻

Reply:
```suggestion
    function onMouseOver(event: MouseEvent) {
```

I haven't tested this, but I think this suggestion is closer to what you're looking for

it might be `: MouseEvent<HTMLButtonElement>` or whatever you HTML element is 🤷 

you'll also need to add the import

```
import React, { MouseEvent } from 'react';
```

https://felixgerschau.com/react-typescript-onmouseover-event-type/

Reply:
appears to have worked... thanks